src/test/java/io/usethesource/vallang/specification/TypeTest.java

```diff
@@ -36,7 +37,7 @@ public void randomTypeGeneratorTestIfTypeStoreContainsGeneratedTypes(TypeStore s
     @ParameterizedTest @ArgumentsSource(ValueProvider.class)
     public void isomorphicStringTest(TypeFactory tf, TypeStore store, Type t) throws IOException {
         // no support for parameter types, aliases and tuple field names yet
-        assertTrue(tf.fromString(store, new StringReader(t.toString())) == t);
+        assertSame(tf.fromString(store, new StringReader(t.toString())), t);
```

Comment:
Does this test the same as `==`? was there a specific reason for `==` to be used instead of `.equals`?